### PR TITLE
4 - Login / Signup API integration

### PR DESCRIPTION
**Pull request by splantebulldozer**
### What it does
* Integrates the frontend signin/up form with the backend auth API
* Error management

### What to look for
* Correct API consumption and error management
* Code style

### What to avoid
* Makes use of the CRA proxy feature to avoid CORS erorrs when running locally, backend icebox issue created @ https://github.com/hatchways/team-bulldozer-1/issues/31
* Redirect to dashboard and user storage to be implemented.

### CODE REVIEW

**Comment by splantebulldozer** [comment]
Errors UI (MUI Snackbars)

image

File: client/src/utils/api/AuthApi.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const API_URL = process.env.REACT_APP_API_URL;
+const FORM_URLENCODED_CONTENT_TYPE = 'application/x-www-form-urlencoded;charset=UTF-8';
+
+export default class AuthApi {
+  static login(username, password) {
+    return axios({
+      url: `${API_URL}/auth/login`,
+      method: 'POST',
+      headers: {
+        'Content-Type': FORM_URLENCODED_CONTENT_TYPE,
```
Comment (skassam21) [inline client/src/utils/api/AuthApi.js]: Is there any reason we are not sending JSON data and sending the username and password in the body of the request object?

Comment (splantebulldozer) [reply client/src/utils/api/AuthApi.js]: It's to comply w/ the API spec that asks for a urlencoded body for the login/register routes.

image

I don't know the specific reaons why @f3rland designed it as such, but from experience it's not uncommon to find auth routes to have this content type, especially when they want to keep the possibility to be directly accessed via the action field of a form tag or when they were in the past and then migrated to being accessed via ajax calls in a SPA.

Comment (f3rland) [reply client/src/utils/api/AuthApi.js:13]: Dit it that way because it's a common way for auth. Can change it to make API uniform

Comment (f3rland) [reply client/src/utils/api/AuthApi.js]: They can actually be passed as json object
```
POST {{host}}/auth/register HTTP/1.1
Content-Type: application/json

{
  "username": "foo@example.com",
  "password": "password"
}
```

**Review by f3rland — APPROVED** [review]

File: client/src/utils/api/AuthApi.js (hunk 1 above)
Comment (splantebulldozer) [reply client/src/utils/api/AuthApi.js]: Reverted to plain ol' JSON and it works.
https://media.giphy.com/media/5aLrlDiJPMPFS/giphy.gif

**Pull request merged**